clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1790,6 +1803,15 @@ private void reportError(
     } else {
       ingestionNotificationDispatcher.reportError(pcsList, message, consumerEx);
     }
+    // Set the replica state to ERROR so that the controller can attempt to reset the partition.
+    if (!isDaVinciClient && resetErrorReplicaEnabled) {
```

Comment:
Based on the title of this PR I'm assuming that for now we don't want to perform reset for non-current version. If that's the case then I don't think this will work for the following reasons:
1. Most caller of this method (e.g. in `processConsumerActions`) will call this only when the SIT version hasn't report completed yet. Meaning it's a future/rebalancing version and not a current serving version and reset is actually undesirable.
2. This method is also called from ST timeout in `StateModelIngestionProgressNotifier`. Again, not a currently serving version and we probably shouldn't reset it.